Создание платежного счета

licosty opened the pull request:
[Задача JIRA](https://faang-school.atlassian.net/browse/BJS2-83925)

licosty committed:
account: controller, service, entity + tests

avzag1 commented on src/main/java/faang/school/accountservice/controller/AccountController.java:
```diff
@@ -0,0 +1,59 @@
+@RestController
+@RequestMapping("/v1/accounts")
+@RequiredArgsConstructor
+public class AccountController {
```
Неплохо бы использовать @ResponseBody

licosty replied:
@RestController уже содержит @responsebody

avzag1 commented on src/main/java/faang/school/accountservice/entity/Account.java:
```diff
@@ -0,0 +1,76 @@
+
+    @Enumerated(EnumType.ORDINAL)
+    @Column(name = "account_type", nullable = false)
+    private AccountType accountType;
```
Лучше использовать EnumType String, чтобы избежать проблем при добавлении новых элементов enum.

licosty replied:
Да, согласна. String звучит лучше

avzag1 commented on src/main/java/faang/school/accountservice/service/AccountService.java:
```diff
@@ -0,0 +1,112 @@
+        account.setNumber(generateAccountNumber());
+
+        Account newAccount = accountRepository.save(account);
+        log.info("New account was created: id={}", newAccount.getId());
```
Хорошо бы вынести в уровень debug

licosty replied:
Я хотела бы знать, что аккаунт успешно создался. Это важная информация, поэтому думаю info лучше

avzag1 commented on src/main/java/faang/school/accountservice/service/AccountService.java:
```diff
@@ -0,0 +1,112 @@
+                .orElseThrow(() -> new EntityNotFoundException("Account not found by id={}", id));
+    }
+
+    @Transactional(readOnly = true)
```
Для чего здесь аннотация?

licosty replied:
Она сообщает Hibernate, что метод используется только для чтения данных из бд. Hibernate видит это и не отслеживает возможные изменения сущности, что ускоряет работу

licosty committed:
account:
- изменение EnumType на STRING в Account;
- миграция для типов столбцов
- корректировка тестов

licosty committed:
account: close, block methods - retryable added

zaqqaz226 commented on src/main/java/faang/school/accountservice/service/AccountService.java:
```diff
@@ -0,0 +1,121 @@
+    }
+
+
+    @Transactional(readOnly = true)
```
А надо тут транзакцию то открывать вообще?)

licosty replied:
Нет, в этих случаях действительно излишне

zaqqaz226 commented on src/main/java/faang/school/accountservice/service/AccountService.java:
```diff
@@ -0,0 +1,121 @@
+    }
+
+    @Transactional
+    @Retryable(retryFor = {OptimisticLockException.class},
```
Мое почтение!)

zaqqaz226 commented:
Мне фактически не к чему придраться(

licosty committed:
account: убраны лишнии транзакции
